### PR TITLE
feat(autofix): Add Q&A interaction

### PR DESCRIPTION
**Pull request by roaga**
Gives Autofix the option to ask the user a question, and continue analysis once they respond.

Also adds a way to disable interactive features when requesting Autofix to start (good for evals, copilot)

### CODE REVIEW

File: src/seer/automation/autofix/models.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ class AutofixStatus(enum.Enum):
     PROCESSING = "PROCESSING"
     NEED_MORE_INFORMATION = "NEED_MORE_INFORMATION"
     CANCELLED = "CANCELLED"
+    WAITING_FOR_USER_RESPONSE = "WAITING_FOR_USER_RESPONSE"
```
Comment (jennmueng) [inline src/seer/automation/autofix/models.py:45]: Do we need both `NEED_MORE_INFORMATION` and `WAITING_FOR_USER_RESPONSE`?

File: src/seer/automation/autofix/components/coding/component.py [hunk 2]
```diff
@@ -52,12 +53,16 @@ def invoke(self, request: CodingRequest) -> CodingOutput | None:
...
+            (
+                CodingPrompts.format_fix_discovery_msg(
+                    event=request.event_details.format_event(),
+                    task_str=task_str,
+                    summary=request.summary,
+                    repo_names=[repo.full_name for repo in state.request.repos],
+                    instruction=request.instruction,
+                )
+                if not request.initial_memory
+                else ""
```
Comment (jennmueng) [inline src/seer/automation/autofix/components/coding/component.py:65]: Shouldn't be passing an empty string here but simply just pass `None`, this way we wouldn't need the `if content` in `add_user_message` and instead just skip adding a user message when the run argument is none

**Review by jennmueng — COMMENTED** [review]
quick first pass, will take a look again in a sec after standup

File: src/seer/automation/autofix/models.py (hunk 1 above)
Comment (roaga) [reply src/seer/automation/autofix/models.py:45]: `NEED_MORE_INFORMATION` is for root cause selection, so I think they should be separate as the behavior once you get a response is very different

File: src/seer/automation/autofix/components/root_cause/component.py [hunk 3]
```diff
@@ -31,31 +31,35 @@ def invoke(
...
-                    repo_names=[repo.full_name for repo in state.request.repos],
+                (
+                    RootCauseAnalysisPrompts.format_default_msg(
+                        event=request.event_details.format_event(),
+                        summary=request.summary,
+                        instruction=request.instruction,
+                        repo_names=[repo.full_name for repo in state.request.repos],
+                    )
+                    if not request.initial_memory
+                    else ""
```
Comment (jennmueng) [inline src/seer/automation/autofix/components/root_cause/component.py:56]: same as last comment above about the run call

**Review by jennmueng — APPROVED** [review]
only that one point otherwise lgtm

File: src/seer/automation/autofix/components/root_cause/component.py (hunk 3 above)
Comment (roaga) [reply src/seer/automation/autofix/components/root_cause/component.py]: Done

File: src/seer/automation/autofix/components/coding/component.py (hunk 2 above)
Comment (roaga) [reply src/seer/automation/autofix/components/coding/component.py]: Done

**Pull request merged**